### DISCUSSION TITLE
Custom DITA-OT error messages

### DISCUSSION THREAD

**Discussion by PieterjanVandenweghe** [dita-ot/dita-ot#4222] [Q&A]
Hello,

I would like to create a validation check thats checks if `//othermeta[@path-portal]` is found in the input file in temporary directory?
If the element is not found, I would like to get an output error message.

What is the best approach to add this to my  DITA-OT plug-ins? 

Thanks a lot in advance for your help.

Kind regards,
Pieterjan

**Comment by jelovirt** [comment]
1. Create a plug-in that uses the extension point `depend.preprocess.post`, similar to https://www.dita-ot.org/dev/topics/plugin-antpreprocess.html.
2. In your custom target, run XSLT to that processes every topic. See e.g. https://github.com/dita-ot/dita-ot/blob/develop/src/main/plugins/org.dita.base/build_preprocess_template.xml#L412 for reference.
3. In the XSLT, output the message configured as in example https://www.dita-ot.org/dev/topics/plugin-messages.html

**Comment by PieterjanVandenweghe** [comment]
Hello Jarno,

Thanks a lot! I got it already working for error messages type ERROR. 
I would like to break the OT generation now.
I thought I could that with the level FATAL, but I get the following error-message:
 [pipeline] Message level  not supported
 
 This is my error message. What am I doing wrong here?

```
<?xml version="1.0" encoding="UTF-8"?>
<messages>
<!-- See resources/messages.xml for the details. -->
<message id="SNCB01F" type="FATAL">
<reason>The build failed due to incorrect DITAMAP.</reason>
<response/>
</message>
</messages>
```
